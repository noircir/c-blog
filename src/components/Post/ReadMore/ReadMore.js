import React from 'react';
import moment from 'moment';
import styles from './ReadMore.module.scss';

const ReadMoreLink = ({ post: { frontmatter: { date, description, slug, title } } }) => (
  <div>
    <a href={slug}><b>{title}</b></a>
    <p><b>{moment(date).format('MMMM D, YYYY')}</b></p>
    <p>{description}</p>
  </div>
);

const ReadMore = ({ prevPost, nextPost }) => (
  <div className={styles['readmore']}>
    <h4 className={styles['readmore-title']}>YOU MIGHT ALSO LIKE</h4>
    <div className={styles['readmore-links']}>
      <ReadMoreLink post={prevPost} />
      <ReadMoreLink post={nextPost} />
      </div>
  </div>
);

export default ReadMore;
