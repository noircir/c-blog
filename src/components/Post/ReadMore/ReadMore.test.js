import React from 'react';
import renderer from 'react-test-renderer';
import ReadMore from './ReadMore';

const post = () => ({
  frontmatter: { date: new Date('3/14/19') },
});

describe('ReadMore', () => {
  it('renders correctly', () => {
    const props = {
      nextPost: post(),
      prevPost: post(),
    };

    const tree = renderer.create(<ReadMore {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
