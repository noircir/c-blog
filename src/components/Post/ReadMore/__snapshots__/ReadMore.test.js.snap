// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReadMore renders correctly 1`] = `
<div
  className="readmore"
>
  <h4
    className="readmore-title"
  >
    YOU MIGHT ALSO LIKE
  </h4>
  <div
    className="readmore-links"
  >
    <div>
      <a>
        <b />
      </a>
      <p>
        <b>
          March 14, 2019
        </b>
      </p>
      <p />
    </div>
    <div>
      <a>
        <b />
      </a>
      <p>
        <b>
          March 14, 2019
        </b>
      </p>
      <p />
    </div>
  </div>
</div>
`;
